src/Humanizer/Localisation/NumberToWords/SpanishNumberToWordsConverter.cs

```diff
@@ -1,193 +1,383 @@
+        {
+            if (number is > 0 and < 10)
             {
-                if (number < 30)
+                string[] ordinalsRootMap = { "" , "primer", "segund", "tercer", "cuart", "quint", "sext", "séptim",
+                "octav", "noven"};
+
+                Dictionary<GrammaticalGender, string> genderedEndingDict = new()
                 {
-                    if (gender == GrammaticalGender.Feminine && (number == 1 || number == 21))
-                    {
-                        parts.Add(number == 1 ? Feminine1 : Feminine21);
-                    }
-                    else
-                    {
-                        parts.Add(UnitsMap[number]);
-                    }
+                    { GrammaticalGender.Feminine, "a" },
+                    { GrammaticalGender.Masculine, HasOrdinalAbbreviation(number, wordForm) ? string.Empty : "o" },
+                };
+
+                genderedEndingDict.Add(GrammaticalGender.Neuter, genderedEndingDict[GrammaticalGender.Masculine]);
+
+                return ordinalsRootMap[number] + genderedEndingDict[gender];
+            }
+
+            return string.Empty;
+        }
+
+        private static string ConvertTenths(in int number, out int remainder, GrammaticalGender gender)
+        {
+            string[] tenthsRootMap = {
+                "", "décim", "vigésim", "trigésim", "cuadragésim", "quincuagésim", "sexagésim", "septuagésim",
+                "octogésim", "nonagésim" };
+
+            return ConvertMappedOrdinalNumber(number, 10, tenthsRootMap, out remainder, gender);
+        }
+
+        private static string ConvertThousandths(in int number, out int remainder, GrammaticalGender gender)
+        {
+            string[] thousandthsRootMap = {
+                "", "milésim", "dosmilésim", "tresmilésim", "cuatromilésim", "cincomilésim", "seismilésim",
+                "sietemilésim", "ochomilésim", "nuevemilésim" };
+
+            return ConvertMappedOrdinalNumber(number, 1000, thousandthsRootMap, out remainder, gender);
+        }
+
+        private static string ConvertUnits(long inputNumber, GrammaticalGender gender, WordForm wordForm = WordForm.Normal)
+        {
+            var genderedOne = new Dictionary<GrammaticalGender, string>()
+            {
+                { GrammaticalGender.Feminine, "una" },
+                { GrammaticalGender.Masculine, wordForm == WordForm.Abbreviation ? "un" : "uno" }
+            };
+
+            genderedOne.Add(GrammaticalGender.Neuter, genderedOne[GrammaticalGender.Masculine]);
+
+            var genderedtwentyOne = new Dictionary<GrammaticalGender, string>()
+            {
+                { GrammaticalGender.Feminine, "veintiuna" },
+                { GrammaticalGender.Masculine, wordForm == WordForm.Abbreviation ? "veintiún" : "veintiuno" }
+            };
+
+            genderedtwentyOne.Add(GrammaticalGender.Neuter, genderedtwentyOne[GrammaticalGender.Masculine]);
+
+            string[] unitsMap = {
```

Comment:
Here

Reply:
This one was a bit trickier for me since I was "injecting" variations for number 1 and 21 in the method.
It is now a static member and I have slightly modified the `ConvertUnits` method to improve the readability.
I hope you like the new approach.